http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeServiceDbQueries.scala

```diff
@@ -52,47 +77,56 @@ object RuntimeServiceDbQueries {
+
+    runtimeQueryFilteredByLabelAndJoinedWithRuntimeAndPatchAndDisk.result.map { x =>
+      val runtimeLabelMap
+        : Map[(ClusterRecord, RuntimeConfig, Option[PatchRecord], Option[DiskConfig]), Map[String, Chain[String]]] =
         x.toList.foldMap {
-          case (((clusterRec, labelRecOpt), runTimeConfigRecOpt), patchRecOpt) =>
+          case ((((runtimeRec, labelRecOpt), runtimeConfigRec), patchRecOpt), diskRecOpt) =>
             val labelMap = labelRecOpt.map(labelRec => labelRec.key -> Chain(labelRec.value)).toMap
-            Map((clusterRec, runTimeConfigRecOpt.map(_.runtimeConfig), patchRecOpt) -> labelMap)
+            val diskConfigOpt = diskRecOpt
+              .map(d => persistentDiskQuery.unmarshalPersistentDisk(d, Map.empty))
+              .map(DiskConfig.fromPersistentDisk)
+            Map((runtimeRec, runtimeConfigRec.runtimeConfig, patchRecOpt, diskConfigOpt) -> labelMap)
         }
 
-      clusterLabelMap.map {
-        case ((clusterRec, runTimeConfigRecOpt, patchRecOpt), labelMap) =>
+      runtimeLabelMap.map {
+        case ((runtimeRec, runtimeConfig, patchRecOpt, diskConfigOpt), labelMap) =>
           val lmp = labelMap.mapValues(_.toList.toSet.headOption.getOrElse(""))
 
           val patchInProgress = patchRecOpt match {
             case Some(patchRec) => patchRec.inProgress
             case None           => false
           }
           ListRuntimeResponse2(
-            clusterRec.id,
-            RuntimeInternalId(clusterRec.internalId),
-            clusterRec.clusterName,
-            clusterRec.googleProject,
-            clusterRec.auditInfo,
-            runTimeConfigRecOpt
-              .getOrElse(
-                throw new Exception(s"No runtimeConfig found for cluster with id ${clusterRec.id}")
-              ), //In theory, the exception should never happen because it's enforced by db foreign key
+            runtimeRec.id,
+            RuntimeInternalId(runtimeRec.internalId),
+            runtimeRec.clusterName,
+            runtimeRec.googleProject,
+            runtimeRec.auditInfo,
+            runtimeConfig,
             Runtime.getProxyUrl(Config.proxyConfig.proxyUrlBase,
-                                clusterRec.googleProject,
-                                clusterRec.clusterName,
+                                runtimeRec.googleProject,
+                                runtimeRec.clusterName,
                                 Set.empty,
                                 lmp),
-            RuntimeStatus.withName(clusterRec.status),
+            RuntimeStatus.withName(runtimeRec.status),
             lmp,
-            patchInProgress
+            patchInProgress,
+            diskConfigOpt
           )
       }.toList
     }
   }
+
+  def isDiskAttachedToRuntime(disk: PersistentDisk)(implicit ec: ExecutionContext): DBIO[Boolean] =
+    clusterQuery.filter(x => x.persistentDiskId.isDefined && x.persistentDiskId === disk.id).length.result.map(_ > 0)
```

Comment:
Reason #2032452346 I hate slick: you use `===` to compare optional and non-optional values